processing/data_collection/gazette/spiders/pr_foz_do_iguacu.py

```diff
@@ -0,0 +1,65 @@
+import datetime as dt
+
+from scrapy.http import FormRequest
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class PrFozDoIguacuSpider(BaseGazetteSpider):
+    MUNICIPALITY_ID = '4108304'
+    BASE_URL = 'http://www.pmfi.pr.gov.br'
+    name = 'pr_foz_do_iguacu'
+    allowed_domains = ['pmfi.pr.gov.br']
+    start_urls = [f'{BASE_URL}/utilidades/diario/index.xhtml']
+
+    def parse(self, response):
+        """
+        @url http://www.pmfi.pr.gov.br/utilidades/diario/index.xhtml
+        @returns requests 1 1
+        """
+        selector = '(//span[@class="ui-paginator-current"])[1]/text()'
+        paginator_text = response.xpath(selector)
+        quantity_of_documents = int(paginator_text.re_first('\([\d]+ de ([\d]+)\)')) + 1
+
+        data = {
+            'formDiarioOficialView': 'formDiarioOficialView',
+            'formDiarioOficialView:dtdiario': 'formDiarioOficialView:dtdiario',
+            'formDiarioOficialView:dtdiario_encodeFeature': 'true',
+            'formDiarioOficialView:dtdiario_first': '0',
+            'formDiarioOficialView:dtdiario_pagination': 'false',
+            'formDiarioOficialView:dtdiario_rows': str(quantity_of_documents),
+        }
+        return FormRequest(response.url, formdata=data, callback=self.parse_items)
+
+    def parse_items(self, response):
+        """
+        @url http://www.pmfi.pr.gov.br/utilidades/diario/index.xhtml
+        @returns items 10 10
+        """
+        lines = response.xpath('//tr[@role="row"]')
+        for line in lines:
```

Comment:
Why not `for line response.xpath('//tr[@role="row"]'):` ?

Reply:
Seems like a personal taste @rennerocha. The way he wrote I think is more readable/organized, for instance.